src/CacheManager.Core/ICacheManager.cs

```diff
@@ -611,6 +611,28 @@ public interface ICacheManager<TCacheValue> : ICache<TCacheValue>
         bool TryUpdate(string key, string region, Func<TCacheValue, TCacheValue> updateValue, int maxRetries, out TCacheValue value);
 
         /// <summary>
+        /// Gets all keys that match the given pattern.
```

Comment:
If it works with having the implementation of the different overloads in BaseCacheHandle, that's good, yes. You'r right, methods must be public ofc, just make public overloads with region cannot accept null as a value